### ISSUE TITLE
[Improvement // Sugestão] Aplicar tipos para as tabs

### ISSUE THREAD

**Issue by EikkoMass** [filipedeschamps/tabnews.com.br#1278]
((Essa issue é inteiramente uma sugestão, sendo aberta para levantar questionamento e uma possibilidade de desenvolvimento))

Acabei reparando que existe uma grande variação de utilizações para as tabs da plataforma, mas nem em todos os casos o modelo estrutural parece suficiente para suprir, em alguns casos podendo ser simples demais, em outros podendo ser complexo demais.

---

**Tipos**

**Fórum:** https://www.tabnews.com.br/userbash/google-apps-scripts

> Esse eu considerando o mais importante, pois seria em grande parte quase nenhum conteúdo agregado à tab, possuindo apenas uma pergunta, que pode acabar possuindo N respostas, com a unica forma do publicador da dúvida indicar qual resolveu seu problema sendo um comentário como "resolveu aqui, obrigado", que muitas vezes pode passar despecebido no possível oceano de respostas recebidas.   

**Artigo:**  https://www.tabnews.com.br/caio/como-exatamente-funciona-a-internet-e-como-esse-post-chegou-ate-voce-conteudo-tecnico-mastigado

> Esse sendo totalmente encaixável ao modelo de tab que possui-se hoje, altamente formatável, com o único ponto a se comentar talvez sendo a agregação do público nos comentários em relação ao mesmo.

**Relato Pessoal:** https://www.tabnews.com.br/Jhonnyssom/meus-primeiros-passos-em-programacao

> Esse acaba trazendo um conteúdo extenso e relativamente adequado ao modelo de hoje, mas talvez eliminando a possibilidade de fonte, pois dificilmente seria agregado com base em algum notíciário ou site externo.  

**Compartilhamento de Cursos / softwares** https://www.tabnews.com.br/maion/conteudo-em-portugues-sobre-aprendizagem-profunda-deep-learning

> Esse se trataria principalmente de compartilhamento de links e base descritiva do mesmos (quando é feita), logo, poderia possuir um template para a inserção de links e a respectiva descrição, ou algo do gênero.

--- 
Obrigado desde já pela possibilidade de sugerir algo do tipo a um projeto como esse.

**Comment by Ryannnkl** [comment]
realmente é algo que aparenta ser bagunçado no feed... acho que um sistema de tags no menu onde a gente possa filtrar por:

- artigo
- duvida
- dicas
- relato
- etc...

resolveria parte desse problema. 
> caso concordem posso começar a ver a possibilidade de implementar tags agora mesmo

**Comment by hkotsubo** [comment]
Concordo com o que @Ryannnkl  [disse acima](https://github.com/filipedeschamps/tabnews.com.br/issues/1278#issuecomment-1446728928). Para a separação que vc propôs, tags me parecem mais adequadas do que criar várias abas, além de ser mais flexível. Por exemplo, eu posso fazer um artigo sobre carreira (com as tags "artigo" e "carreira"), mas também posso fazer um relato pessoal sobre carreira (aí teria as tags "relato" e "carreira"), etc. Isso ajuda também a categorizar melhor os posts (desde que coloque um limite razoável, 5 ou 6 tags no máximo - mais que isso começa a poluir demais e vira ruído).

Inclusive, isso abre a possibilidade de fazer buscas por tags, o que diminui bastante a necessidade de várias abas. Talvez poderia ter abas fixas para tags mais comuns, não sei.

Mas em vez de abas, eu sugiro algo mais geral, como a possibilidade de ordenar posts por critérios diferentes (podendo ser crescente ou decrescente):

- Últimas Atualizações: posts que tiveram alguma atualização mais recente (comentário, edição, etc)
- Pontuação: ordenar por posts por pontuação
- Quantidade de comentários
- Etc...

Com isso, juntamente com as tags e um sistema de busca, qualquer um pode filtrar e ordenar os posts da maneira que quiser. Indo um pouco além, cada usuário poderia salvar seus próprios filtros personalizados.

**Comment by aprendendofelipe** [comment]
@EikkoMass, obrigado pela sugestão! 💪

Vou fechar a issue por ser relacionada com outra já aberta, então recomendo centralizar a discussão na #742.

Combinado? 🤝
